src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder finalInfo = new StringBuilder(
+                String.format("Report for period %s - %s", dateFrom, dateTo))
+                .append(System.lineSeparator());
+
+        LocalDate from = LocalDate.parse(dateFrom, DATE_FORMAT);
+        LocalDate to = LocalDate.parse(dateTo, DATE_FORMAT);
+
+        for (int i = 0; i < names.length; i++) {
+            int salaryInfo = 0;
+            for (int j = 0; j < data.length; j++) {
+                if (data[j].contains(names[i])) {
```

Comment:
lets split this line as well, you will have a bug if there would be similar names, for ex. Jonh Jonathan